src/libcrun/linux.c

```diff
@@ -1684,17 +1687,22 @@ libcrun_create_dev (libcrun_container_t *container, int devfd, int srcfd,
         }
       else
         {
-          char *dirname;
-          cleanup_free char *buffer = NULL;
           cleanup_close int dirfd = -1;
-          char *basename, *tmp;
+          cleanup_free char *dirname = NULL;
+          char *basename, *found;
 
-          buffer = xstrdup (device->path);
-          dirname = buffer;
+          dirname = strdup (normalized_path);
 
-          tmp = strrchr (buffer, '/');
-          *tmp = '\0';
-          basename = tmp + 1;
+          found = strrchr (dirname, '/');
+          if (found)
+            {
+              *found = '\0';
+              basename = found + 1;
+            }
+          else
+            {
+              basename = dirname;
+            }
```

Comment:
Would this be the correct behaviour here?

Reply:
This ensures that if there are no slashes in the path, `basename` is set to the entire path. This prevents any uninitialized variables and potential segmentation faults. Just an improvement over the original code. We need to clarify with @kolyshkin or @giuseppe to understand if there's any repercussion to it.

Reply:
Example Scenarios:

   -  **Path with Slashes:**
        **Input:** "/path/to/file.txt"
        **Operation:**
            found = strrchr(dirname, '/') will point to the last slash ('/').
            *found = '\0' will truncate dirname to "/path/to".
            basename = found + 1 will set basename to "file.txt" (the part after the last slash).
        **Result:**
            dirname = "/path/to"
            basename = "file.txt"
   
 - **Path Without Slashes:**
        **Input:** "file.txt"
        **Operation:**
            found = strrchr(dirname, '/') will return NULL since there are no slashes in the path.
            The line basename = found ? found + 1 : dirname; will evaluate to basename = dirname;, meaning basename will be set to "file.txt".
        **Result:**
            dirname = "file.txt"
            basename = "file.txt"
   
-  **Single Slash:**
        **Input:** "/"
        **Operation:**
            found = strrchr(dirname, '/') will point to the single slash.
            *found = '\0' will truncate dirname to an empty string ("").
            basename = found ? found + 1 : dirname; will evaluate to basename = ""; (because found + 1 points to the character after the null terminator).
        **Result:**
            dirname = ""
            basename = ""
- **Empty String:**
        **Input**: ""
        **Operation**:
            found = strrchr(dirname, '/') will return NULL.
            basename = found ? found + 1 : dirname; will evaluate to basename = ""; (because dirname is empty).
        **Result:**
            dirname = ""
            basename = ""